packages/lambda-tiler/src/routes/attribution.ts

```diff
@@ -0,0 +1,211 @@
+ */
+function roundNumber(n: number): number {
+    return Math.round(n * Precision) / Precision;
+}
+
+function roundPair(p: Pair): Pair {
+    return [roundNumber(p[0]), roundNumber(p[1])];
+}
+
+/**
+ * Convert a list of COG file bounds into a MultiPolygon. If the bounds spans more than half the
+ * globe then return a simple MultiPolygon for the bounding box.
+
+ * @param bbox in WGS84
+ * @param files in target projection
+ * @return MultiPolygon in WGS84
+ */
+function createCoordinates(bbox: BBox, files: NamedBounds[], proj: Projection): MultiPolygon {
+    if (Wgs84.delta(bbox[0], bbox[2]) <= 0) {
+        // This bounds spans more than half the globe which multiPolygonToWgs84 can't handle; just
+        // return bbox as polygon
+        return Wgs84.bboxToMultiPolygon(bbox);
+    }
+
+    let coordinates: MultiPolygon = [];
+
+    // merge imagery bounds
+    for (const image of files) {
+        const poly = [Bounds.fromJson(image).pad(SmoothPadding).toPolygon()] as MultiPolygon;
+        coordinates = union(coordinates, poly);
+    }
+
+    const roundToWgs84 = (p: number[]): number[] => roundPair(proj.toWgs84(p) as Pair);
+
+    return multiPolygonToWgs84(coordinates, roundToWgs84);
+}
+
+async function readStac(uri: string): Promise<StacCollection | null> {
+    try {
+        return await FileOperator.readJson<StacCollection>(uri);
+    } catch (err) {
+        if (FileOperator.isCompositeError(err) && err.code < 500) {
+            return null;
+        }
+        throw err;
+    }
+}
+
+/**
+ * Build a Single File STAC for the given TileSet.
+ *
+ * For now this is the minimal set required for attribution. This can be embellished later with
+ * links and assets for a more comprehensive STAC file.
+ */
+async function tileSetAttribution(tileSet: TileSet): Promise<AttributionStac | null> {
+    const proj = Projection.get(tileSet.projection.code);
+    const stacFiles = new Map<string, Promise<StacCollection | null>>();
+    const cols: AttributionCollection[] = [];
+    const items: AttributionItem[] = [];
+
+    // read all stac files in parallel
+    for (const rule of tileSet.tileSet.rules) {
+        const im = tileSet.imagery.get(rule.imgId);
+        if (im == null) continue;
+        if (stacFiles.get(im.uri) == null) {
+            stacFiles.set(im.uri, readStac(FileOperator.join(im.uri, 'collection.json')));
+        }
+    }
+
+    const host = await Aws.tileMetadata.Provider.get(TileMetadataTag.Production);
+    if (host == null) return null;
+
+    for (const rule of tileSet.tileSet.rules) {
+        const im = tileSet.imagery.get(rule.imgId);
+        if (im == null) continue;
+        const stac = await stacFiles.get(im.uri);
+
+        const bbox = proj.boundsToWgs84BoundingBox(im.bounds).map(roundNumber) as BBox;
+
+        let interval: [string, string][] | undefined = stac?.extent.temporal?.interval;
+        if (interval == null) {
+            const years = extractYearRangeFromName(im.name);
```

Comment:
do we have any stac files that need this? I would prefer not to guess names here, we should just fix the STAC file

Reply:
It is for when we don't have a stac file; in practice we should have one but hard to guarantee.

Reply:
I can remove the `?` in `temporal?`; it should not be null.

Reply:
Can we just not have a date? Id prefer not to have this code here, we should fix the stac files if they are broken.